test/com/google/enterprise/adaptor/DocIdSenderTest.java

```diff
@@ -491,6 +534,56 @@ public void testAclItemToString() {
     assertEquals(golden, "" + new DocIdSender.AclItem(id, acl));
   }
 
+
+  private static Map<GroupPrincipal, Collection<Principal>> groupsSample() {
+    // Order of iteration matters
+    Map<GroupPrincipal, Collection<Principal>> groups
+        = new TreeMap<GroupPrincipal, Collection<Principal>>();
+    groups.put(new GroupPrincipal("g1"),
+        Arrays.asList(new UserPrincipal("u1"), new GroupPrincipal("g2")));
+    groups.put(new GroupPrincipal("g2"),
+        Arrays.asList(new UserPrincipal("u2"), new GroupPrincipal("g3")));
+    groups.put(new GroupPrincipal("g3"),
+        Arrays.asList(new UserPrincipal("u3"), new GroupPrincipal("g4")));
+    return Collections.unmodifiableMap(groups);
+  }
+
+  private static List<List<Map.Entry<GroupPrincipal, Collection<Principal>>>>
+      unsplitExpectedResult() {
+    List<List<Map.Entry<GroupPrincipal, Collection<Principal>>>> goldenGroups
+        = new ArrayList<List<Map.Entry<GroupPrincipal,
+          Collection<Principal>>>>();
+    List<Map.Entry<GroupPrincipal, Collection<Principal>>> tmp
+        = new ArrayList<Map.Entry<GroupPrincipal, Collection<Principal>>>();
+    tmp.add(new SimpleImmutableEntry<GroupPrincipal, Collection<Principal>>(
+        new GroupPrincipal("g1"), SAMPLE_DATA.get(new GroupPrincipal("g1"))));
+    tmp.add(new SimpleImmutableEntry<GroupPrincipal, Collection<Principal>>(
+        new GroupPrincipal("g2"), SAMPLE_DATA.get(new GroupPrincipal("g2"))));
+    tmp.add(new SimpleImmutableEntry<GroupPrincipal, Collection<Principal>>(
+        new GroupPrincipal("g3"), SAMPLE_DATA.get(new GroupPrincipal("g3"))));
+    goldenGroups.add(tmp);
+    return goldenGroups;
+  }
+
+  private static List<List<Map.Entry<GroupPrincipal, Collection<Principal>>>>
+      splitPer2GroupsExpectedResult() {
```

Comment:
What about Lists.partion(unsplitExpectedResult().get(0), 2), or something of that nature, or building both off just the inner list?